ledger-core/virtual/handlers/handle_meta.go

```diff
@@ -33,78 +35,79 @@ type errNoHandler struct {
 	messageType   reflect.Type
 }
 
-func FactoryMeta(message *statemachine.DispatcherMessage) (pulse.Number, smachine.CreateFunc) {
-	payloadMeta := message.PayloadMeta
-	messageMeta := message.MessageMeta
+func FactoryMeta(msg *statemachine.DispatcherMessage, authService authentication.Service, pr pulse.Range) (pulse.Number, smachine.CreateFunc, error) {
```

Comment:
1. we should convert that Factory to object (not to pass authService in every message) and embed pulse.Range into DispatcherMessage